src/data/list/perm.lean

```diff
@@ -880,6 +880,125 @@ begin
   { refine (IH₁ H).trans (IH₂ ((p₁.pairwise_iff _).1 H)),
     exact λ a b h h₁ h₂, h h₂ h₁ }
 end
+lemma perm.take_inter {α} [decidable_eq α] {xs ys : list α} (n : ℕ)
+  (h : xs ~ ys) (h' : ys.nodup) :
+  xs.take n ~ ys.inter (xs.take n) :=
+begin
+  simp only [list.inter] at *,
+  induction h generalizing n,
+  case list.perm.nil
+  { simp only [not_mem_nil, filter_false, take_nil] },
+  case list.perm.cons
+  { cases n; simp only [mem_cons_iff, true_or, eq_self_iff_true, filter_cons_of_pos, perm_cons, take, not_mem_nil,
```

Comment:
Could you re-wrap the long lines in this file so that they're < 100 chars?

I'm a bit more concerned about the length of the proofs here and how repetitive they seem (admittedly I didn't study them too hard yet). Are there some missing lemmas or tactics that could help here?

Reply:
I have found induction on permutation harder to decompose into lemmas. Admittedly, I didn't try very hard